webui/react/src/pages/TaskList.tsx

```diff
@@ -58,178 +57,34 @@ interface SourceInfo {
   sources: TensorBoardSource[];
 }
 
-const defaultFilters: TaskFilters<CommandType> = {
-  limit: MINIMUM_PAGE_SIZE,
-  states: undefined,
-  types: undefined,
-  users: undefined,
-};
-
-const defaultSorter: ApiSorter = {
-  descend: true,
-  key: 'startTime',
-};
-
-const URL_ALL = 'all';
-
-const STORAGE_PATH = 'task-list';
-const STORAGE_FILTERS_KEY = 'filters';
-const STORAGE_LIMIT_KEY = 'limit';
-const STORAGE_SORTER_KEY = 'sorter';
-
 const TaskList: React.FC = () => {
   const { users } = useStore();
-  const storage = useStorage(STORAGE_PATH);
-  const initLimit = storage.getWithDefault(STORAGE_LIMIT_KEY, MINIMUM_PAGE_SIZE);
-  const initFilters = storage.getWithDefault(STORAGE_FILTERS_KEY, { ...defaultFilters });
-  const [ isUrlParsed, setIsUrlParsed ] = useState(false);
+  const { filterCount, resetFilters, settings, updateSettings } = useSettings<Settings>(
+    settingsConfig,
+    paths.taskList(),
```

Comment:
The hook maintains its own state. Looking at code again, if the path changes, the only thing that gets updated is the `updateSettings` callback. The existing local storage and the user settings state shouldn't get affected.

Reply:
> if this path or the way we generate it changes in any way it'll blow away user settings.

I think I get what you're putting down. Like `/tasks` getting changed to `/task-list`, then URLs that use `/tasks` won't work.

Reply:
> > if this path or the way we generate it changes in any way it'll blow away user settings.
> 
> I think I get what you're putting down. Like `/tasks` getting changed to `/task-list`, then URLs that use `/tasks` won't work.

yeah this one

Reply:
maybe if our `paths` returned the full route instead of just the resolved url we could have used the route id which would be less likely to change but a constant works great as well IMO